pkg/sloop/store/typed/eventcounttablegen_test.go

```diff
@@ -48,5 +48,67 @@ func Test_ResourceEventCountsTable_SetWorks(t *testing.T) {
 		return nil
 	})
 	assert.Nil(t, err)
+}
+
+func helper_update_ResourceEventCountsTable(t *testing.T, keys []string, val *ResourceEventCounts) (badgerwrap.DB, *ResourceEventCountsTable) {
+	untyped.TestHookSetPartitionDuration(time.Hour)
+	b, err := (&badgerwrap.MockFactory{}).Open(badger.DefaultOptions(""))
+	assert.Nil(t, err)
+	wt := OpenResourceEventCountsTable()
+	err = b.Update(func(txn badgerwrap.Txn) error {
+		var txerr error
+		for _, key := range keys {
+			txerr = wt.Set(txn, key, val)
+			if txerr != nil {
+				return txerr
+			}
+		}
+		// Add some keys outside the range
+		txerr = txn.Set([]byte("/a/123/"), []byte{})
+		if txerr != nil {
+			return txerr
+		}
+		txerr = txn.Set([]byte("/zzz/123/"), []byte{})
+		if txerr != nil {
+			return txerr
+		}
+		return nil
+	})
+	assert.Nil(t, err)
+	return b, wt
+}
+
+func Test_ResourceEventCountsTable_GetUniquePartitionList_Success(t *testing.T) {
+	if helper_ResourceEventCounts_ShouldSkip() {
+		return
+	}
 
+	db, wt := helper_update_ResourceEventCountsTable(t, (&EventCountKey{}).SetTestKeys(), (&EventCountKey{}).SetTestValue())
```

Comment:
because the test is not looking at the stored value, please use `ResourceEventCounts{}` instead of `(&EventCountKey{}).SetTestValue()`